src/components/StarRatingContainer.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, createElement } from "react";
+import { StarRating } from "./StarRating";
+
+interface WrapperProps {
+    class?: string;
+    mxObject?: mendix.lib.MxObject;
+    style?: string;
+}
+
+export interface ContainerProps extends WrapperProps {
+    editable: "default" | "never";
+    mxObject: mendix.lib.MxObject;
+    viewAverage: boolean;
+    readOnly: boolean;
+    // Properties from Mendix modeler
+    rateAttribute: string;
+    onChangeMicroflow: string;
+}
+
+interface ContainerState {
+    initialRate: number;
+}
+
+export default class StarRatingContainer extends Component<ContainerProps, ContainerState> {
+    private subscriptionHandles: number[];
+
+    constructor(props: ContainerProps) {
+        super(props);
+
+        this.subscriptionHandles = [];
+        this.state = {
+            initialRate: this.props.mxObject
+                ? this.props.mxObject.get(this.props.rateAttribute) as number
+                : 0
+        };
+        this.handleOnChange = this.handleOnChange.bind(this);
+        this.subscribe(this.props.mxObject);
+    }
+
+    render() {
+        const { mxObject } = this.props;
+        const readOnly = this.props.editable === "never"
+            || (mxObject && mxObject.isReadonlyAttr(this.props.rateAttribute)) || this.props.readOnly || !mxObject;
+
+        return createElement(StarRating, {
+            className: this.props.class,
+            fractions: readOnly ? 2 : 1,
```

Comment:
Fractions should not be part of the container logic; it is display logic, only mx interactions should be in the container. Move this settings to the _StarRating.ts_